src/App.tsx

```diff
@@ -1,37 +1,96 @@
-import React from 'react';
-import Header from './components/Header/Header';
+import './App.scss'
+
+import React, {useMemo, createContext, useEffect} from 'react';
 import TodoList from './components/TodoList/TodoList';
 import AddTodo from './components/AddTodo/AddTodo';
-import { Box } from '@mui/material';
-import { makeStyles, createStyles } from '@mui/styles';
+import {Box, IconButton, Card, CardHeader, CardContent, CardActions} from '@mui/material';
+import {ThemeProvider, createTheme} from '@mui/material/styles';
+import Brightness4Icon from '@mui/icons-material/Brightness4';
+import Brightness7Icon from '@mui/icons-material/Brightness7';
+import useMediaQuery from '@mui/material/useMediaQuery';
 
-const useStyles = makeStyles(() =>
-  createStyles({
-    appWrapper: {
-      backgroundColor: '#f8f9fa',
-      height: '100vh'
-    },
-    todoListWrapper: {
-      margin: '0 auto',
-      border: '1px solid black',
-      borderRadius: '5px',
-      minHeight: '50vh',
-      width: '100%',
-      maxWidth: 360,
-    }
-  })
-);
+import {switchTheme, saveState, LIGHT, DARK} from './redux/appSlice';
+import {useDispatch, useSelector} from 'react-redux';
+import {RootState} from './redux/store';
+
+const ColorModeContext = createContext({
+  toggleColorMode: (mode: 'light' | 'dark' | undefined) => {
+  }
+});
 
 const App = () => {
-  const classes = useStyles();
+  const dispatch = useDispatch();
+  const mode = useSelector((state: RootState) => state.config.mode);
+  const prefersColorScheme = useMediaQuery('(prefers-color-scheme: dark)') ? DARK : LIGHT;
+  useEffect(() => {
+    if (!mode) {
+      dispatch(switchTheme(prefersColorScheme));
+    }
+  }, []);
+  useEffect(() => {
+    if (mode) {
+      saveState(mode);
+    }
+  }, [mode]);
+  const colorMode = useMemo(
+    () => ({
+      toggleColorMode: (mode: 'light' | 'dark' | undefined) => {
+        dispatch(switchTheme(mode === LIGHT ? DARK : LIGHT));
+      },
+    }),
+    [],
+  );
+
+  const theme = useMemo(
+    () =>
+      createTheme({
+        palette: {
+          mode,
+        },
+        typography: {
+          fontFamily: 'Inter, sans-serif',
+        },
+      }),
+    [mode],
+  );
+
   return (
-    <div className={classes.appWrapper}>
-      <Header title={'ToDo List'} />
-      <Box className={classes.todoListWrapper}>
-        <AddTodo />
-        <TodoList />
-      </Box>
-    </div>
+    <ColorModeContext.Provider value={colorMode}>
```

Comment:
Where is the consumer? It seems like you don't need this context at all

Reply:
In the Material UI docs, the 'ColorModeContext.Provider' is used without the 'Consumer' too:
https://mui.com/material-ui/customization/dark-mode/#system-ToggleColorMode.tsx